spec/services/unwithdraw_service_spec.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+RSpec.describe UnwithdrawService do
+  before do
+    @user = create(:user)
+  end
+
+  describe "#process" do
+    it "adds an entry in the timeline of the document" do
+      withdrawn_edition = create(:edition, :withdrawn)
+      stub_publishing_api_republish(withdrawn_edition.content_id, {})
+
+      UnwithdrawService.new.process(withdrawn_edition, @user)
+
+      expect(withdrawn_edition.timeline_entries.first.entry_type).to eq("unwithdrawn")
+    end
+
+    it "creates a new status with the same state as the previously published version" do
+      withdrawn_edition = create(:edition, :withdrawn)
+      stub_publishing_api_republish(withdrawn_edition.content_id, {})
+
+      previously_published_state = withdrawn_edition.status.details.published_status.state
+
+      UnwithdrawService.new.process(withdrawn_edition, @user)
+      withdrawn_edition.reload
+
+      expect(withdrawn_edition.status.state).to eq(previously_published_state)
+    end
+
+    it "raises an error if current edition is published" do
+      published_edition = create(:edition, :published)
+      stub_publishing_api_republish(published_edition.content_id, {})
+
+      expect { UnwithdrawService.new.process(published_edition, @user) }.to raise_error(
+        "edition must be withdrawn to be unwithdrawn"
+      )
+    end
+
+    it "raises an error if current edition is draft" do
```

Comment:
minor but the condition here is probably that we care that the item is "not withdrawn" rather than it's specifically a "draft"

Reply:
Yep i've removed this test now as its actually just a duplicate of the one above